src/main/scala/com/fulcrumgenomics/umi/ConsensusCallingIterator.scala

```diff
@@ -40,7 +40,7 @@ import com.fulcrumgenomics.util.ProgressLogger
   * @param caller the consensus caller to use to call consensus reads
   * @param progress an optional progress logger to which to log progress in input reads
   * @param threads the number of threads to use.
-  * @param chunkSize parallel process in chunkSize units; will cause 8 * chunkSize records to be held in memory
+  * @param chunkSize across the input [[SamRecord]]s from this many source molecules at a time
```

Comment:
The new param doc doesn't track for me.  Based on my comment below:
```suggestion
  * @param chunkSize the number of source molecules to process in a single chunk, where chunks
                     are processed in parallel.  This will cause `2 * threads * chunkSize` source 
                     molecules to be held in memory, where the number of [[SamRecord]]s
                      held in memory depends on the number of [[SamRecord]]s assigned to 
                      each source molecule at any given time.
```

Reply:
If this is true, then presumably in `CallDuplexConsensusReads` the number given should be divided before supplying here?  I think it will be very confusing to say "only hold 10 source molecules worth of reads in memory" and then find out that with 8 threads I've got 160 molecules worth.

Reply:
@tfenne @nh13 Many of these comments, here and elsewhere in the review, go against my interpretation of how `ParIterator` works. The way I understand it, these iterable functions operate on a single chunk at a time (not counting `.toAsync`), using the multi-threaded task support, rather then grabbing one chunk per-thread:

https://github.com/fulcrumgenomics/commons/blob/9372c5585084a1d7486c1b6036d0e06824b2c8d4/src/main/scala/com/fulcrumgenomics/commons/collection/ParIterator.scala#L140-L141

Only one chunk is pulled into memory at a time at the "front" of the `ParIterator` (it does not mention `threads * chunkSize`:

https://github.com/fulcrumgenomics/commons/blob/9372c5585084a1d7486c1b6036d0e06824b2c8d4/src/main/scala/com/fulcrumgenomics/commons/collection/ParIterator.scala#L119-L120

The `parWith` function splits a single chunk across threads:

https://github.com/fulcrumgenomics/commons/blob/9372c5585084a1d7486c1b6036d0e06824b2c8d4/src/main/scala/com/fulcrumgenomics/commons/CommonsDef.scala#L360-L364

It was based on this interpretation that I do not divide by `threads` before parallelizing. Can you please let me know if I have missed something or where my interpretation is wrong?